aisynphys/data/data.py

```diff
@@ -337,17 +338,56 @@ def _get_tserieslist(self, ts_name, align, bsub, bsub_win=5e-3):
                 else:
                     baseline = float_mode(baseline_data)
                 ts = ts - baseline
-            
+
             if align is not None:
-                if align == 'spike':
-                    align_t = pr.stim_pulse.first_spike_time
-                    # ignore PRs with no known spike time
-                    if align_t is None:
+                align_times = {
+                    'spike': {'pulse': pr.stim_pulse.first_spike_time,
+                             'average': np.mean([p.stim_pulse.first_spike_time for p in self.prs if p.stim_pulse.first_spike_time is not None])},
+                    'peak': {'pulse': pr.stim_pulse.spikes[0].peak_time if pr.stim_pulse.n_spikes==1 else None,
+                              'average': np.mean([p.stim_pulse.spikes[0].peak_time for p in self.prs if p.stim_pulse.n_spikes==1 and p.stim_pulse.spikes[0].peak_time is not None])},
+                    'stim': {'pulse': stim_time, 
+                              'average': np.mean([p.stim_pulse.onset_time for p in self.prs if p.stim_pulse.onset_time is not None])},
+                }
```

Comment:
This will be very slow:
1. All three options spike/peak/stim are computed, despite only one of them being kept
2. The average values are being recomputed for every pulse, even though the result will be the same every time